servicebus/data-plane/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/amqp/BaseLinkHandler.java

```diff
@@ -19,19 +19,17 @@ public BaseLinkHandler(final IAmqpLink amqpLink) {
 
     @Override
     public void onLinkLocalClose(Event event) {
-        Link link = event.getLink();
-        if (link != null) {            
+        final Link link = event.getLink();
+        if (link != null) {
             TRACE_LOGGER.debug("local link close. linkName:{}", link.getName());
+            closeSession(link);
```

Comment:
[SpotBugs-P2]
Null passed for non-null parameter of closeSession(Link) in com.microsoft.azure.servicebus.amqp.BaseLinkHandler.onLinkLocalClose(Event)

This method call passes a null value for a non-null method parameter. Either the parameter is annotated as a parameter that should always be non-null, or analysis has shown that it will always be dereferenced.